src/modules/samples/cpp/src/so/SampleModule.cpp

```diff
@@ -0,0 +1,270 @@
+        }
+    }
+    else
+    {
+        OsConfigLogError(SampleLog::Get(), "MmiOpen called with null clientName");
+        status = EINVAL;
+    }
+
+    return handle;
+}
+
+void MmiClose(MMI_HANDLE clientSession)
+{
+    Sample* session = reinterpret_cast<Sample*>(clientSession);
+    if (nullptr != session)
+    {
+        delete session;
+    }
+}
+
+int MmiSet(
+    MMI_HANDLE clientSession,
+    const char* componentName,
+    const char* objectName,
+    const MMI_JSON_STRING payload,
+    const int payloadSizeBytes)
+{
+    int status = MMI_OK;
+    Sample* session = nullptr;
+
+    ScopeGuard sg{[&]()
+    {
+        if (MMI_OK == status)
+        {
+            if (IsFullLoggingEnabled())
+            {
+                OsConfigLogInfo(SampleLog::Get(), "MmiSet(%p, %s, %s, %.*s, %d) returned %d", clientSession, componentName, objectName, payloadSizeBytes, payload, payloadSizeBytes, status);
+            }
+        }
+        else
+        {
+            if (IsFullLoggingEnabled())
+            {
+                OsConfigLogError(SampleLog::Get(), "MmiSet(%p, %s, %s, %.*s, %d) returned %d", clientSession, componentName, objectName, payloadSizeBytes, payload, payloadSizeBytes, status);
+            }
+            else
+            {
+                OsConfigLogError(SampleLog::Get(), "MmiSet(%p, %s, %s, -, %d) returned %d", clientSession, componentName, objectName, payloadSizeBytes, status);
+            }
+        }
+    }};
+
+    if (nullptr == clientSession)
+    {
+        OsConfigLogError(SampleLog::Get(), "MmiSet called with null clientSession");
+        status = EINVAL;
+    }
+    else
+    {
+        session = reinterpret_cast<Sample*>(clientSession);
+        status = session->Set(componentName, objectName, payload, payloadSizeBytes);
+    }
+
+    return status;
+}
+
+int MmiGet(
+    MMI_HANDLE clientSession,
+    const char* componentName,
+    const char* objectName,
+    MMI_JSON_STRING* payload,
+    int* payloadSizeBytes)
+{
+    int status = MMI_OK;
+    Sample* session = nullptr;
+
+    ScopeGuard sg{[&]()
+    {
+        if (IsFullLoggingEnabled())
+        {
+            if (MMI_OK == status)
+            {
+                OsConfigLogInfo(SampleLog::Get(), "MmiGet(%p, %s, %s, %.*s, %d) returned %d", clientSession, componentName, objectName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+            }
+            else
+            {
+                OsConfigLogError(SampleLog::Get(), "MmiGet(%p, %s, %s, %.*s, %d) returned %d", clientSession, componentName, objectName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+            }
+        }
+    }};
+
+    if (nullptr == clientSession)
+    {
+        OsConfigLogError(SampleLog::Get(), "MmiGet called with null clientSession");
+        status = EINVAL;
+    }
+    else
+    {
+        session = reinterpret_cast<Sample*>(clientSession);
+        status = session->Get(componentName, objectName, payload, payloadSizeBytes);
```

Comment:
Add a comment here and in MmiSet how the handle can be different for each separate client session, similar to the comment on line 143